### PR TITLE
feat: Add support for multiple document loader plugins (PDF, YAML, TXT, DOCX)

### PR DESCRIPTION
**Pull request by sajosam**
### Summary:
This PR adds support for loading and processing multiple document types (.pdf, .yaml, .txt, .docx). Uploaded files are validated and loaded into a vector store for further processing.

### Key Changes:
- Created a file upload handler class to support PDF, YAML, TXT, and DOCX file formats.
- Implemented file extension validation to ensure only supported formats are processed.
- Added a maximum file size limit of 10 MB to prevent large uploads.
- Enabled the ability to load data from multiple documents simultaneously into a vector store.

### CODE REVIEW

File: app/api/v1/connector.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 import app.services.connector as svc
 import app.services.provider as provider_svc
 from starlette.requests import Request
+from fastapi import APIRouter, UploadFile, File
```
Comment (sajosam) [inline app/api/v1/connector.py:9]: Import required library for upload documents

File: app/api/v1/connector.py [hunk 2]
```diff
@@ -82,6 +83,39 @@ def get_connector(connector_id: int, db: Session = Depends(get_db)):
         error=None
     )
 
+@router.post("/upload/datasource", response_model=resp_schemas.CommonResponse)
+async def upload_document_datsource(
```
Comment (sajosam) [inline app/api/v1/connector.py:87]: Async route function for upload documents

File: app/api/v1/connector.py [hunk 3]
```diff
@@ -82,6 +83,39 @@ def get_connector(connector_id: int, db: Session = Depends(get_db)):
...
+    Uploads an document data source file to the server.
+
+    Args:
+        file (UploadFile): The uploaded document file. Accepted formats are .pdf, .txt, .yaml, and .docx.
+
+    Returns:
+        CommonResponse: A response containing the file upload status, file details, or an error message.
+    """
+
+    error, size = await svc.fileValidation(file)
```
Comment (sajosam) [inline app/api/v1/connector.py:101]: File validation checker

File: app/api/v1/connector.py [hunk 4]
```diff
@@ -82,6 +83,39 @@ def get_connector(connector_id: int, db: Session = Depends(get_db)):
...
+
+    if error:
+        return commons.is_error_response("Invalid File", error, {"file_path": None})
+    
+    result, error = await svc.upload_pdf(file)
```
Comment (sajosam) [inline app/api/v1/connector.py:106]: Upload file to assets/datasource/documents/<uuid>/<file_name>

File: app/base/document_data_plugin.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
+from abc import ABC, abstractmethod
+from typing import Dict, Any, Optional
+
+class DocumentDataPlugin(ABC):
```
Comment (sajosam) [inline app/base/document_data_plugin.py:4]: Defined abstract method for DocumentDataPlugin

File: app/plugins/document/__init__.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
+from app.models.prompt import Prompt
+from collections import OrderedDict
+from app.models.request import ConnectionArgument
+
+# Plugin Metadata
+__version__ = '1.0.0'
```
Comment (sajosam) [inline app/plugins/document/__init__.py:6]: Defined document plugins base details

File: app/plugins/document/__init__.py [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+__plugin_name__ = 'document'
+__display_name__ = 'document loader'
+__description__ = 'document integration for handling document data'
+__icon__ = '/assets/plugins/logos/document.svg'
+__category__ = 4
+
+# Connection arguments
+__connection_args__ = OrderedDict(
```
Comment (sajosam) [inline app/plugins/document/__init__.py:14]: Defined configuration params for document plugins

File: app/plugins/document/__init__.py [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+        description = 'Supports only .pdf, .yaml, .txt, and .docx files.',
+        order = 1,
+        required = True,
+        value = None,
+        slug = "document_file"
+    )
+)
+
+# Prompt
+__prompt__ = Prompt(**{
```
Comment (sajosam) [inline app/plugins/document/__init__.py:27]: Base prompt for document plugins

File: app/plugins/document/formatter.py [hunk 9]
```diff
@@ -0,0 +1,32 @@
+from typing import Any, Dict
+from loguru import logger
+
+class Formatter:
```
Comment (sajosam) [inline app/plugins/document/formatter.py:4]: Formatter class for document plugins

File: app/plugins/document/handler.py [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+import requests
+from app.base.base_plugin import BasePlugin
+from app.base.remote_data_plugin import RemoteDataPlugin
+from app.base.plugin_metadata_mixin import PluginMetadataMixin
+from app.base.document_data_plugin import DocumentDataPlugin
+from typing import  Tuple, Optional
+from app.readers.base_reader import BaseReader
+
+
+class Document(BasePlugin, PluginMetadataMixin,DocumentDataPlugin,  Formatter):
```
Comment (sajosam) [inline app/plugins/document/handler.py:12]: Handler class for Document

File: app/services/connector.py [hunk 11]
```diff
@@ -751,6 +809,12 @@ def formatting_datasource(connector, provider):
             'params': connector.connector_config,
             'documentations': [{'type': 'text', 'value': connector.connector_docs}]
         }
+    elif provider.category_id == 4:
+        return {
+            'type': provider.key,
+            'params': connector.connector_config,
+            'documentations': [{'type': 'text', 'value': connector.connector_docs}]
+        }
```
Comment (agberoz) [inline app/services/connector.py:817]: connector docs may not be required here, as document type plugins don't have  documentation section

File: app/providers/data_preperation.py [hunk 12]
```diff
@@ -34,7 +34,7 @@ def get_source_documents(self):
 
 
             for docs in self.documentation:
```
Comment (agberoz) [inline app/providers/data_preperation.py:36]: Try running the connector along with a database connector and ensure this conversion doesn't cause any issues

File: app/plugins/document/__init__.py [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+            Adhere to these rules while generating query:
+            - Deliberately go through the question and context word by word to appropriately answer the question
+            """
+        },
+        "user_prompt":{
+            "template": """
+            User question is "$question"
+            generate a json in the following format without any formatting.
+            {
+                "explanation": "Explain how you finalized the sql query using the schemas and rules provided",
```
Comment (agberoz) [inline app/plugins/document/__init__.py:47]: wrong prompt

File: app/plugins/document/handler.py [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+            if not data:
+                raise ValueError("No data fetched during health check")
+            logger.info("Checking documentation")
+            return True, None
+        except Exception as e:
+            logger.exception(f"Exception during fetching data: {str(e)}")
+            return False, str(e)
+
+    def fetch_data(self, params=None):
+        data = []
```
Comment (agberoz) [inline app/plugins/document/handler.py:56]: move this supported type into self.params

**Review by agberoz — CHANGES_REQUESTED** [review]

**Comment by sajosam** [comment]
Link to Demo : https://www.loom.com/share/8b6c0d23bcb542dba526c424d24fc97c?sid=edb5bf30-b1a3-4beb-a454-14dc61dbe9be

**Review by agberoz — APPROVED** [review]

**Review by agberoz — APPROVED** [review]

**Pull request merged**